src/main/java/org/example/springbootintro/controller/BookController.java

```diff
@@ -51,4 +62,10 @@ public void deleteById(@PathVariable Long id) {
     public BookDto updateById(@PathVariable Long id, @RequestBody CreateBookRequestDto bookDto) {
         return bookService.updateById(id, bookDto);
     }
+
+    @GetMapping("/search")
+    @ResponseStatus(HttpStatus.OK)
+    public List<BookDto> search(@Valid BookSearchParametersDto bookSearchParametersDto) {
+        return bookService.search(bookSearchParametersDto);
+    }
```

Comment:
This changes is not related to task

Reply:
Add meaningful validation for existing DTO classes
I add my own validator. To search by title or author or both, and see error message about empty field and so on.